### ISSUE TITLE
Assembly version for System.ValueTuple is incorrect

### ISSUE THREAD

**Issue by weshaggard** [dotnet/corefx#31987]
While investigating something else today @ericstj and I noticed that the assembly version for System.ValueTuple.dll shipped as 4.0.3.0 in 2.1 and as such may cause some assembly-ref-def mismatches on some platforms.

.NET Framework in version 4.7.1 has version 4.0.2.0 inbox, but the latest System.ValueTuple package has references for net461/net47/portable that have version 4.0.3.0. We might be OK because this assembly was added to the unification table in 4.7.1 as well and thus 3rd and 4th part of the version will be ignored when unifying. More testing should be done to determine if there is any real issue we might need to service for.

FYI @jcouv. 

Given this is inbox now we are also planning to discontinue shipping this as an individual library package any longer.

**Comment by weshaggard** [comment]
@Anipik I explicitly tagged as 2.1.x as I think we need to investigate if there is something we should service here.

**Comment by danmosemsft** [comment]
@weshaggard is this still considered for servicing? If so is someone looking at it? If not we should change milestone

**Comment by weshaggard** [comment]
Yes we should consider this for servicing but it needs further investigation from the owner of this library to understand the impact.

**Comment by Anipik** [comment]
> @Anipik I explicitly tagged as 2.1.x as I think we need to investigate if there is something we should service here.

It was removed by the GithubIssueLabeler bot. It sometimes does that, investigating the reason behind it

**Comment by danmosemsft** [comment]
@jcouv do you own ValueTuple?

**Comment by danmosemsft** [comment]
@weshaggard 	can you help get the right owner engaged?

**Comment by danmosemsft** [comment]
@jcouv what was the outcome -- this is still marked 2.1.x -- is there a servicing level change here? If not is there a change needed in master?

**Comment by jcouv** [comment]
So, I ran the validation that Wes had recommended:
1. create a 4.6 or 4.7 library with tuples (I did 4.6.1)
2. consume the resulting binary from a 4.7.1 client program (I did 4.7.2)

When I compile the client, I get `error CS0012: The type 'ValueTuple<,>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.ValueTuple, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.`

The IDE then offers me to add that reference. When I do, then things compile and execute fine.

I believe that is the expected behavior, but I will check with @weshaggard.
